### ISSUE TITLE
Cubic Spline

### ISSUE THREAD

**Issue by felipenoris** [invenia/Impute.jl#15]
Would you like an implementation of Natural Cubic Spline interpolation for missing data?

**Comment by rofinn** [comment]
Absolutely, implementations of other imputation methods are very welcome! Our only requirements are that the new code includes tests and documentation to help with maintainability. 

We'd like to keep this package fairly minimal, so any methods that depend on heavy optimization or ml libraries would be better off in a separate package; dependence on common stats libraries like StatsBase, Distributions or MultivariateStats would be fine though.
